File: frontend/src/app/shared/types/entities/TranslationVersion.ts
export class TranslationVersion {
	id: number;
	content: string;
	locale: string;
	updateDate: Date;
	isValid: boolean;

	createdBy: string;

	originalTranslationId: number;

	constructor(id: number, content: string, locale: string, updateDate: Date, isValid: boolean, createdBy: string, originalTranslationId: number) {
		this.id = id;
		this.content = content;
		this.locale = locale;
		this.updateDate = updateDate;
		this.isValid = isValid;
		this.createdBy = createdBy;
		this.originalTranslationId = originalTranslationId;
	}
}

File: backend/src/main/java/org/tim/entities/TranslationVersion.java
package org.tim.entities;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;

import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;
import java.util.Locale;

@Document(indexName = "translation-version")
@Data
@NoArgsConstructor
public class TranslationVersion {

    @Id
    @Setter(AccessLevel.NONE)
    private String id;

    @NotNull
    private String content;

    @NotNull
    @NonNull
    private Locale locale;

    @JsonSerialize(using = LocalDateTimeSerializer.class)
    private LocalDateTime updateDate;

    private Boolean isValid;

    private String createdBy;

    private Boolean isArchived;

    @NotNull
    private String translationId;
}
